FF_calculation/FF_ttbar.py

```diff
@@ -6,61 +6,263 @@
+
+            h = rdf_AR.Histo1D(
+                (
+                    process_conf["var_dependence"],
+                    f"{sample}",
+                    nbinsx,
+                    xbinning,
+                ),
+                process_conf["var_dependence"],
+                "weight",
+            )
+            AR_hists[sample] = h.GetValue()
+
+    # Start of the FF calculation
+    FF_hist = ff_func.calculate_ttbar_FF(
+        SR=SR_hists,
+        AR=AR_hists,
+        SRlike=SRlike_hists,
+        ARlike=ARlike_hists,
+    )
+    # performing the fit and calculating the uncertainties
+    fit_graphs, corrlib_exp, used_fit = ff_func.fit_function(
+        ff_hists=FF_hist.Clone(),
+        bin_edges=binning,
+        logger=logger,
+        fit_option=process_conf.get("fit_option", gd.default_fit_options["ttbar"]),
+        limit_kwargs=process_conf.get(
+            "limit_kwargs",
+            gd.get_default_fit_function_limit_kwargs(binning),
+        ),
+    )
+
+    plotting.plot_FFs(
+        variable=process_conf["var_dependence"],
+        ff_ratio=FF_hist,
+        uncertainties=fit_graphs,
+        era=config["era"],
+        channel=config["channel"],
+        process=process,
+        category=split,
+        output_path=output_path,
+        logger=logger,
+        draw_option=used_fit,
+        save_data=True
+    )
+
+    # doing some control plots
+    for _hist, _region, _data, _samples in [
+        (SRlike_hists, "SR_like", "data", controlplot_samples(config)),
+        (ARlike_hists, "AR_like", "data", controlplot_samples(config)),
+        (SRlike_hists, "SR_like", "data_subtracted", ["ttbar_J"]),
+        (ARlike_hists, "AR_like", "data_subtracted", ["ttbar_J"]),
+    ]:
+        for yscale, save_data in zip(["linear", "log"], [True, False]):
+            plotting.plot_data_mc_ratio(
+                variable="metphi",
+                hists=_hist,
+                era=config["era"],
+                channel=config["channel"],
+                process=process,
+                region=_region,
+                data=_data,
+                samples=_samples,
+                category=split,
+                output_path=output_path,
+                logger=logger,
+                yscale=yscale,
+                save_data=save_data,
+            )
+    log.info("-" * 50)
+
+    return ff_func.fill_corrlib_expression(corrlib_exp, split_variables, split)
+
+
+def calculation_FF_prerequisites(
     config: Dict[str, Union[str, Dict, List]],
+    process_conf: Dict[str, Union[str, Dict, List]],
     sample_paths: List[str],
-    output_path: str,
     process: str,
     logger: str,
-) -> Dict[str, Union[Dict[str, str], Dict[str, Dict[str, str]]]]:
+) -> Tuple[Dict[str, ROOT.TH1D], Dict[str, ROOT.TH1D]]:
     """
-    This function calculates fake factors for ttbar.
+    This function calculates the global SR and AR histograms for the ttbar process
+    that are used for process normalization.
```

Comment:
I would change the following here:
* "SR-like and AR-like" instead of "SR and AR"
*  at the end "... process normalization to data."